### PR TITLE
[ROCm][CI] Fix custom logits entrypoints for ROCm/XPU spawn

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
Follow-up: #42040

- Centralize test multiprocessing start-method selection in `tests/utils.py`.
- Make custom logits processor entrypoint setup choose fork-compatible monkeypatching or spawn-compatible dist-info registration by platform.
- Update online and offline custom logits processor tests to use the shared entrypoint setup helper.
- Keep CUDA and other fork-capable platforms on the fork path while using spawn on ROCm/XPU.

cc @kenroche

### CODE REVIEW

**Comment by AndreasKaratzas** [comment]
Closing in favor of original PR.
